src/addresses/xaddress.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021 Logos Foundation
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <addresses/xaddress.h>
+#include <base58.h>
+#include <hash.h>
+#include <uint256.h>
+#include <util/strencodings.h>
+#include <util/string.h>
+#include <util/vector.h>
+
+#include <string>
+
+namespace XAddress {
+/**
+ * Create a checksum.
```

Comment:
This is quite confusing. Can you make this return just the 4 bytes?

Also rename to something like "CalcChecksum"

Reply:
Doing that would just involve another copy, but I can do it if you want. Renaming it to that would be incorrect. It's not a checksum, and the existing Base58 library uses the term `Check`

Reply:
Why is it not a checksum? Because it's not reversible?

Reply:
Generally this can't tell you where the error is.

Reply:
Either way, I need to fix the comment.

Reply:
Ok, so how do we call that part then? Just `check`? I don't like that; it implies something different, and is used in other places as "the act of checking".

According to [Wikipedia](https://en.wikipedia.org/wiki/Checksum), a checksum doesn't require to be able to tell where an error is: "The simplest checksum algorithm is the so-called longitudinal parity check", which doesn't tell you where the error is.

I'd prefer if we used the term "checksum", pretty much everyone will know what it means this way.